pkg/verification/sender/twilio_sender_test.go

```diff
@@ -0,0 +1,93 @@
+package sender_test
+
+import (
+	"bytes"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	sender2 "github.com/codeready-toolchain/registration-service/pkg/verification/sender"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+type MockTwilioConfig struct {
+	AccountSID    string
+	AuthToken     string
+	FromNumber    string
+	SenderConfigs []toolchainv1alpha1.TwilioSenderConfig
+}
+
+func (c *MockTwilioConfig) TwilioAccountSID() string {
+	return c.AccountSID
+}
+
+func (c *MockTwilioConfig) TwilioAuthToken() string {
+	return c.AuthToken
+}
+
+func (c *MockTwilioConfig) TwilioFromNumber() string {
+	return c.FromNumber
+}
+
+func (c *MockTwilioConfig) TwilioSenderConfigs() []toolchainv1alpha1.TwilioSenderConfig {
+	return c.SenderConfigs
+}
+
+func TestTwilioSenderIDs(t *testing.T) {
+
+	httpClient := &http.Client{Transport: &http.Transport{}}
+	gock.InterceptClient(httpClient)
+
+	defer gock.Off()
+
+	gock.New("https://api.twilio.com").
+		Reply(http.StatusNoContent).
+		BodyString("")
+
+	var reqBody io.ReadCloser
+	obs := func(request *http.Request, mock gock.Mock) {
+		reqBody = request.Body
+		defer func(Body io.ReadCloser) {
+			err := Body.Close()
+			require.NoError(t, err)
+		}(request.Body)
+	}
+
+	gock.Observe(obs)
+
+	ctx, _ := gin.CreateTestContext(httptest.NewRecorder())
+
+	cfg := &MockTwilioConfig{
+		AccountSID: "TWILIO_SID_VALUE",
+		AuthToken:  "AUTH_TOKEN_VALUE",
+		FromNumber: "+13334445555",
+		SenderConfigs: []toolchainv1alpha1.TwilioSenderConfig{
+			{
+				SenderID:     "RED HAT",
+				CountryCodes: []string{"44"},
+			},
+		},
+	}
+
+	sender := sender2.NewTwilioSender(cfg, httpClient)
+
+	err := sender.SendNotification(ctx, "Test Message", "+440000000000", "44")
```

Comment:
Can we also add a test which would send to a number which doesn't have the country code representation in the config? Basically another notification to any country code but "44". To check it doesn't use the sender ID and not just fail.

Reply:
Done